pkg/ebpf/uprobes/inspector.go

```diff
@@ -0,0 +1,176 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package uprobes
+
+import (
+	"debug/elf"
+	"fmt"
+	"os"
+	"runtime"
+
+	manager "github.com/DataDog/ebpf-manager"
+
+	"github.com/DataDog/datadog-agent/pkg/network/go/bininspect"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/common"
+)
+
+// BinaryInspector implementors are responsible for extracting the metadata required to attach from a binary.
+type BinaryInspector interface {
+	// Inspect returns the metadata required to attach to a binary. The first return is the metadata, the second
+	// return is a boolean indicating whether this binary is compatible and can be attached or not.
+	// It is encouraged to return early if the binary is not compatible, to avoid unnecessary work.
+	// In the future, the first and second return values should be merged into a single struct, but for
+	// now this allows us to keep the API compatible with the existing implementation.
+	Inspect(fpath utils.FilePath, requests []SymbolRequest) (map[string]bininspect.FunctionMetadata, bool, error)
+
+	// Cleanup is called when a certain file path is not needed anymore, the implementation can clean up
+	// any resources associated with the file path.
+	Cleanup(fpath utils.FilePath)
+}
+
+// SymbolRequest represents a request for symbols and associated data from a binary
+type SymbolRequest struct {
+	Name                   string
+	BestEffort             bool
+	IncludeReturnLocations bool
```

Comment:
While it is clear to me, I don't think the average reader understands BestEffort and IncludeReturnLocations meaning
Maybe a documentation can help here